librz/core/linux_heap_glibc.c

```diff
@@ -1485,15 +1575,176 @@ void GH(print_malloc_info)(RzCore *core, GHT m_state, GHT malloc_state) {
 	}
 }
 
+/**
+ * \brief Prints the heap chunks in a bin with double linked list (small|large|unsorted)
+ * \param core RzCore pointer
+ * \param main_arena MallocState struct for the arena in which bins are
+ * \param bin_num The bin number for the bin from which chunks have to printed
+ * \return number of chunks found in the bin
+ */
+static int GH(print_bin_content)(RzCore *core, MallocState *main_arena, int bin_num) {
+	int idx = 2 * bin_num;
+	ut64 fw = main_arena->GH(bins)[idx];
+	ut64 bk = main_arena->GH(bins)[idx + 1];
+
+	GH(RzHeapChunk) *head = RZ_NEW0(GH(RzHeapChunk));
+	if (!head) {
+		return 0;
+	}
+	(void)rz_io_read_at(core->io, bk, (ut8 *)head, sizeof(GH(RzHeapChunk)));
+
+	size_t chunks_cnt = 0;
+	if (head->fd == fw) {
+		return chunks_cnt;
+	}
+
+	switch (bin_num) {
+	case 0:
+		rz_cons_printf("Unsorted");
+		break;
+	case 1 ... NSMALLBINS - 1:
```

Comment:
I think we can't use such thing, as it is GNU specific AFAIK. Please just use `if`/`else` instead.